test/Interop/Cxx/namespace/Inputs/namespaces.h

```diff
@@ -0,0 +1,73 @@
+namespace BasicParent {
+namespace BasicChild {
+void basicTestFunction() {}
+void forwardDeclared();
+void definedInDefs();
+} // namespace BasicChild
+} // namespace BasicParent
+
+namespace BasicParent {
+namespace BasicChild {
+void inRedecl() {}
+} // namespace BasicChild
+} // namespace BasicParent
+
+namespace BasicParent {
+namespace BasicChild {
+void forwardDeclared() {}
+} // namespace BasicChild
+} // namespace BasicParent
+
+namespace BasicParent {
+namespace BasicChild {
+namespace SecondLevelChild {
+void inSubNamespaceInRedecl() {}
+} // namespace SecondLevelChild
+} // namespace BasicChild
+} // namespace BasicParent
+
+namespace UniqueOuterName1 {
+namespace SameInnerName {
+void basicTestFunction() {}
+} // namespace SameInnerName
+} // namespace UniqueOuterName1
+
+namespace UniqueOuterName2 {
+namespace SameInnerName {
+void basicTestFunction() {}
+} // namespace SameInnerName
+} // namespace UniqueOuterName2
+
+namespace HasStructs {
+struct BasicStruct {
+  void basicTestFunction() {}
+};
+struct ForwardDeclaredStruct;
+} // namespace HasStructs
+
+struct HasStructs::ForwardDeclaredStruct {};
+
+namespace ParentWithChildWithStructs {
+namespace ChildWithStructs {
+struct ForwardDeclaredChildStruct;
+struct DefinedInDefsStruct;
+} // namespace ChildWithStructs
+} // namespace ParentWithChildWithStructs
+
+namespace ParentWithChildWithStructs {
+struct ChildWithStructs::ForwardDeclaredChildStruct {
+  void basicTestFunction() {}
+};
+} // namespace ParentWithChildWithStructs
+
+namespace ParentWithChildWithFunctions {
+namespace ChildWithFunctions {
+void forwardDeclared();
+void definedInDefs();
+void basicTestFunction() {}
+} // namespace ChildWithFunctions
+} // namespace ParentWithChildWithFunctions
+
+namespace ParentWithChildWithFunctions {
+void ChildWithFunctions::forwardDeclared(){};
+}
```

Comment:
Note to self: we need to change the way we import class template specializations, now (and add tests for that change).